src/features/transfer/TransfersDetailsModal.tsx

```
@@ -56,15 +68,112 @@ export function TransfersDetailsModal({
+    isOpen,
+    multiProvider,
+  );
+
+  // Determine message stage from delivery data + transfer status
+  // We use our own logic instead of the widget's useMessageStage to avoid
+  // broken Explorer REST API dependencies (queryExplorerForBlock, /latest-nonce)
+  const stage = useMemo((): MessageStage => {
+    if (delivery.isDelivered) return MessageStage.Relayed;
+    if (isSent && originTxHash) return MessageStage.Sent;
+    return MessageStage.Preparing;
+  }, [delivery.isDelivered, isSent, originTxHash]);
```

ESLint

[error] Compilation Skipped: Existing memoization could not be preserved

React Compiler has skipped optimizing this component because the existing manual memoization could not be preserved. This dependency may be mutated later, which could cause the value to change unexpectedly.

/home/jailuser/git/src/features/transfer/TransfersDetailsModal.tsx:107:29
  105 |     if (isSent && originTxHash) return MessageStage.Sent;
  106 |     return MessageStage.Preparing;
> 107 |   }, [delivery.isDelivered, isSent, originTxHash]);
      |                             ^^^^^^ This dependency may be modified later
  108 |
  109 |   // Resolve the destination tx hash from either store or live query
  110 |   const destinationTxHash = storedDestTxHash || delivery.destinationTxHash;

(react-hooks/preserve-manual-memoization)